quarkus/integration-tests/integration-tests-quarkus-legacy-rules/src/main/java/org/kie/kogito/legacy/rules/TmsEndpoint.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.legacy.rules;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+
+import org.kie.api.runtime.ClassObjectFilter;
+import org.kie.api.runtime.KieRuntimeBuilder;
+import org.kie.api.runtime.KieSession;
+
+@Path("/test-tms")
+public class TmsEndpoint {
+    @Inject
+    KieRuntimeBuilder kieRuntimeBuilder;
+
+    @POST()
+    public Object executeQuery(@QueryParam("string") String string) {
+        return test(kieRuntimeBuilder, string);
+    }
+
+    public static Object test(KieRuntimeBuilder kieRuntimeBuilder, String string) {
+        KieSession session = kieRuntimeBuilder.newKieSession();
+
+        session.insert(string);
+        session.fireAllRules();
+
+        return session.getObjects(new ClassObjectFilter(Integer.class)).iterator().next();
+    }
```

Comment:
This code is not testing if the `logicalInsert` is working as expected (aka remove the string -> remove the int).
I suggest to do something like (note: pseudo code)
```suggestion
    KieSession session;  // initialised via @PostConstruct or similar

    @GET()
    public Object executeQuery() {
        return test();
    }
    
    @POST()
    public Object insertString(@QueryParam("string") String string) {
        session.insert(string);
        session.fireAllRules();

        return session.getObjects(new ClassObjectFilter(Integer.class)).iterator().next();
    }
    
    @DELETE()
    public Object deleteString(@QueryParam("string") String string) {
        session.remove(string);
        session.fireAllRules();

        return session.getObjects(new ClassObjectFilter(Integer.class)).iterator().next();
    }

    public static Object test() {
        return session.getObjects(new ClassObjectFilter(Integer.class)).iterator().next();
    }
```

And then do a couple of calls to see the different return values

Reply:
@danielezonca I changed the test to actually check the logical assertion.